flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -227,13 +241,19 @@ public void startNewWorker(ResourceProfile resourceProfile) {
 	}
 
 	@Override
-	public void stopWorker(ResourceID resourceID) {
-		// TODO: Implement to stop the worker
+	public void stopWorker(YarnWorkerNode workerNode) {
+		workerNodeMap.remove(workerNode.getResourceID().toString());
+		if (workerNode != null) {
+			log.info("Stopping container {}.", workerNode.getYarnContainer().getId().toString());
+			resourceManagerClient.releaseAssignedContainer(workerNode.getYarnContainer().getId());
```

Comment:
I think we should also call `nodeManagerClient.stopContainer()` before.

Reply:
Good point. In such case, I think we need YarnWorkerNode to keep the returned Container in order to retrieve both the containerId and nodeId to call stopContainer.

Reply:
So I think it's necessary to have the workerNodeMap to keep the mapping from resourceId to YarnWorkerNode as resourceId does not have the nodeId information. What do you think?

Reply:
Yes, I think that's also how we do it in the old `YarnFlinkResourceManager`.